### PR TITLE
chore: release 2.4.1

### PR DESCRIPTION
**Pull request by plamut**
This is a release PR with polished release notes (supersedes #346).

### CODE REVIEW

**Review by anguillanneuf — APPROVED** [review]

**Comment by anguillanneuf** [comment]
I think the release notes in the release bot-proposed PR are editable. In Java at least, the edits in the samples directory are not included in the release notes.

**Comment by plamut** [comment]
@anguillanneuf Are you saying that if one edits the release PR's description, that those changes also get reflected in `CHANGELOG.md`? That would indeed be quite useful.

**Comment by plamut** [comment]
Since I might be partially offline tomorrow and won't be able to jump in immediately if anything breaks, I would postpone merging this until Tuesday afternoon or Wednesday morning, unless someone has free cycles to be on alert?

cc: @pradn

**Comment by anguillanneuf** [comment]
@plamut I believe so. I never touched `change.md` in the Java repos I know and it has been self-updating based on commit messages.

**Comment by plamut** [comment]
@anguillanneuf Ah, what I meant was amending the notes if commit messages themselves are incomplete or contain typos (it happens occasionally). One then needs to update `CHANGELOG.md` manually, or...?

**Comment by anguillanneuf** [comment]
@plamut Updating the PR description should be enough. We can try too. Published release notes can be edited as well in the release notes page if something slips through unintended.

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,27 @@
 [1]: https://pypi.org/project/google-cloud-pubsub/#history
 
 
+## [2.4.1](https://www.github.com/googleapis/python-pubsub/compare/v2.4.0...v2.4.1) (2021-03-24)
```
Comment (plamut) [inline CHANGELOG.md:8]:
```suggestion
## [2.4.1](https://www.github.com/googleapis/python-pubsub/compare/v2.4.0...v2.4.1) (2021-03-30)
```

**Pull request merged**